skore/src/skore/sklearn/_estimator/feature_importance_accessor.py

```diff
@@ -1,16 +1,150 @@
+    "log_loss": make_scorer(metrics.log_loss),
+    "r2": make_scorer(metrics.r2_score),
+    "rmse": make_scorer(metrics.root_mean_squared_error),
+}
+
+
+def _check_scoring(scoring: Any) -> Union[Scoring, None]:
+    """Check that `scoring` is valid, and convert it to a suitable form as needed.
+
+    If `scoring` is a list of strings, it is checked against our own metric names.
+    For example, "rmse" is recognized as root-mean-square error, even though sklearn
+    itself does not recognize this name.
+    Similarly, "neg_root_mean_square_error" is not recognized, and leads to an error.
+
+    Parameters
+    ----------
+    scoring : str, callable, list, tuple, dict, or None
+        The scoring to check.
+
+    Returns
+    -------
+    scoring
+        A scoring hopefully suitable for passing to `permutation_importance`.
+        Can be equal to the original scoring.
+
+    Raises
+    ------
+    TypeError
+        If `scoring` does not type-check.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import make_scorer, root_mean_squared_error
+    >>> from skore.sklearn._estimator.feature_importance_accessor import _check_scoring
+
+    >>> _check_scoring(None)  # Returns None
+
+    >>> _check_scoring(make_scorer(root_mean_squared_error))
+    make_scorer(root_mean_squared_error, ...)
+
+    >>> _check_scoring({"rmse": make_scorer(root_mean_squared_error)})
+    {'rmse': make_scorer(root_mean_squared_error, ...)}
 
+    >>> _check_scoring("rmse")
+    {'rmse': make_scorer(root_mean_squared_error, ...)}
 
-class _FeatureImportanceAccessor(_BaseAccessor["EstimatorReport"], DirNamesMixin):
+    >>> _check_scoring(["r2", "rmse"])
+    {'r2': make_scorer(r2_score, ...),
+    'rmse': make_scorer(root_mean_squared_error, ...)}
+
+    >>> _check_scoring("neg_root_mean_squared_error")
+    Traceback (most recent call last):
+    TypeError: If scoring is a string, it must be one of ...;
+    got 'neg_root_mean_squared_error'
+
+    >>> _check_scoring(["r2", make_scorer(root_mean_squared_error)])
+    Traceback (most recent call last):
+    TypeError: If scoring is a list or tuple, it must contain only strings; ...
+
+    >>> _check_scoring(3)
+    Traceback (most recent call last):
+    TypeError: scoring must be a string, callable, list, tuple or dict;
+    got <class 'int'>
+    """
+    if scoring is None or callable(scoring) or isinstance(scoring, dict):
+        return scoring
+    elif isinstance(scoring, str):
+        if scoring in metric_to_scorer:
+            # Convert to scorer
+            return {scoring: metric_to_scorer[cast(Metric, scoring)]}
+        raise TypeError(
+            "If scoring is a string, it must be one of "
+            f"{list(metric_to_scorer.keys())}; got '{scoring}'"
+        )
+    elif isinstance(scoring, (list, tuple)):
+        result: dict[str, Callable] = {}
+        for s in scoring:
+            if isinstance(s, str):
+                result |= cast(dict[str, Callable], _check_scoring(s))
```

Comment:
I would advocate for this explicit call.

```suggestion
                result.update(
                    cast(dict[str, Callable], _check_scoring(s)
                )
```

Reply:
i like it like this, it makes it clear that we are mutating the dict

Reply:
The assumption here is that you are used to this operator but at the long term, depending on who will read this code (and the seniority), `update` is more explicit. But I'm not sure that I'll convince you :).

Reply:
I agree in principle (we had this debate with @thomass-dev about `:=`) but in this case I don't find `update` much more explicit. I would expect a new Python developer to google both `update` and `|` equally often